addons/stock/models/stock_rule.py

```diff
@@ -350,40 +354,49 @@ class ProcurementGroup(models.Model):
         required=True)
 
     @api.model
-    def run(self, procurements):
+    def run(self, procurements, raise_if_exception=True):
         """ Method used in a procurement case. The purpose is to supply the
         product passed as argument in the location also given as an argument.
         In order to be able to find a suitable location that provide the product
         it will search among stock.rule.
         """
```

Comment:
this docstring is so sad

``` python
"""Fulfil `procurements` with the help of stock rules.

Procurements are needs of products at a certain location. To fulfil
these needs, we need to create some sort of documents (`stock.move`
by default, but extensions of `_run_` methods allow to create every
type of documents).

:param procurements: the description of the procurement
:type list: list of `~odoo.addons.stock.models.stock_rule.ProcurementGroup.Procurement`
:param raise_if_exception: will raise either an UserError or a ProcurementException
:type raise_if_exception: boolan, optional
:raises UserError: if `raise_if_exception` is True and a procurement isn't fulfillable
:raises ProcurementException: if `raise_if_exception` is False and a procurement isn't fulfillable
"""
```
note: raise_if_exception=False still make the method raise